src/systems/core/utils.ts

```diff
@@ -0,0 +1,21 @@
+import { enterQuery, exitQuery, IWorld } from 'bitecs';
+import { InitializeComponentDataProps } from './types';
+
+export function createInitializeData({ componentRef, query, data, factory }: InitializeComponentDataProps) {
+    const enter = enterQuery(query);
+    const exit = exitQuery(query);
+
+    return (world: IWorld) => {
```

Comment:
return (world: IWorld) => {
    for (const entity of enter(world)) {
        const newIndex = data.length;
        data = [...data, factory(entity)];
        componentRef[entity] = newIndex;
    }

    for (const entity of exit(world)) {
        const last = data[data.length - 1];
        data = data.slice(0, -1); 

        if (componentRef[entity] === data.length) continue;

        componentRef[last.entity] = componentRef[entity];
        data[componentRef[entity]] = last;
    }
};

Reply:
Esse utilitário serve justamente para mutacionar o data... isso ai já quebraria toda a idéia do utilitário.
Sem falar do problema do garbage collector.

O slice é muito anti-performatico, o utilitário existe justamente para evitar isso também